kolibri/core/errorreports/test/test_middleware.py

```diff
@@ -0,0 +1,16 @@
+from unittest.mock import patch
+
+from django.test import TestCase
+
+from ..middleware import ErrorReportingMiddleware
+from ..models import ErrorReports
+
+
+class ErrorReportingMiddlewareTestCase(TestCase):
+    @patch.object(ErrorReports, "insert_or_update_error")
+    def test_process_exception(self, mock_insert_or_update_error):
+        middleware = ErrorReportingMiddleware(lambda r: None)
+        request = self.client.request()
+        exception = Exception("Test Exception")
+        middleware.process_exception(request, exception=exception)
+        mock_insert_or_update_error.assert_called_once()
```

Comment:
We could also use `assert_called_once_with(...)` to check if the mock was called with the expected arguments. where ... are the expected arguments

Reply:
We can also simulate an integrity error using the side_effect argument eg

```python
@patch.object(ErrorReports, "insert_or_update_error", side_effect=IntegrityError)
@patch.object(logging, 'getLogger')
def test_process_exception_with_integrity_error(self, mock_insert_or_update_error, mock_logging)
    mock_logging.assert_called_once_with("Expected error message")
```

Reply:
We could also test the success case too

Reply:
> @patch.object(ErrorReports, "insert_or_update_error", side_effect=IntegrityError)
@patch.object(logging, 'getLogger')
def test_process_exception_with_integrity_error(self, mock_insert_or_update_error, mock_logging)
    mock_logging.assert_called_once_with("Expected error message")


[This](https://github.com/learningequality/kolibri/pull/12260/files#diff-09250651e602f41122ea1b5cd5579c05a28b5555e966106f1b3f7d0a86463c2aR34) shall be the `Expected error message` ?

Reply:
> We could also test the success case too

By mocking logger right?

Reply:
if you are interested in the messages, but it could be just if the insert_or_update() has been called with the right arguments. There are a few permutations here you could consider 🙂

Reply:
Ok I will see what i can do